packages/transports/aws-cloudwatch-logs/src/strategies/default.strategy.ts

```diff
@@ -1,31 +1,53 @@
-import { CloudWatchLogsClient, type InputLogEvent } from "@aws-sdk/client-cloudwatch-logs";
-import { ensureGroupExists, ensureStreamExists, sendEvents } from "../actions.js";
-import type { CloudWatchLogsStrategy, CloudWatchLogsStrategyOptions, ICloudWatchLogsStrategy } from "./common.js";
+import { CloudWatchLogsClient, type CloudWatchLogsClientConfig } from "@aws-sdk/client-cloudwatch-logs";
+import type { SendEventParams } from "../common.types.js";
+import { ensureGroupExists, ensureStreamExists, sendEvents } from "../utils.js";
+import { BaseStrategy } from "./base.strategy.js";
 
-// The default strategy is responsible for handling log events and sending them to CloudWatch Logs.
-class DefaultStrategy implements ICloudWatchLogsStrategy {
-  #clientInstance?: CloudWatchLogsClient;
+export interface DefaultCloudWatchStrategyStrategyOptions {
+  /**
+   * AWS CloudWatch Logs client configuration to use when creating a new client.
+   */
+  clientConfig?: CloudWatchLogsClientConfig;
 
-  constructor(protected readonly options: CloudWatchLogsStrategyOptions) {}
+  /**
+   * Try to create the log group and log stream if they don't exist yet when sending logs.
+   * @defaultValue false
+   */
+  createIfNotExists?: boolean;
```

Comment:
i moved this property out of the common interface because a strategy implementer might not care to add support for creating streams